Modules/timemodule.c

```diff
@@ -2114,8 +2139,9 @@ pysleep(_PyTime_t secs)
 #endif
 
         /* sleep was interrupted by SIGINT */
-        if (PyErr_CheckSignals())
+        if (PyErr_CheckSignals()) {
             return -1;
+        }
 
         if (get_monotonic(&monotonic) < 0) {
```

Comment:
Recomputing secs is useless if clock_nanosleep() is used no? It might be more efficient to avoid the get_monotonic() syscall.

Reply:
Yes, recomputing of monotonic time is totally useless if clock_nanosleep() is used. Moreover the do {} while loop is usless too in this case.